### ISSUE TITLE
NullReferenceException in GenerateRuntimeConfigurationFiles.AddFrameworks

### ISSUE THREAD

**Issue by sharwell** [dotnet/sdk#3857]
Error occurred building xunit/xunit.analyzers@74f6d366bd5a33f7ed966fd6f1fb7703452dac0d with VS 16.4 Preview 3.

I'm not sure what the issue here is, but it seems possibly related to #3697.

```
C:\Program Files\dotnet\sdk\3.1.100-preview2-014569\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(255,5): error MSB4018: The "GenerateRuntimeConfigurationFiles" task failed unexpectedly. [C:\dev\github\xunit.analyzers\test\xunit.analyzers.tests\xunit.analyzers.tests.csproj]
C:\Program Files\dotnet\sdk\3.1.100-preview2-014569\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(255,5): error MSB4018: System.NullReferenceException: Object reference not set to an instance of an object. [C:\dev\github\xunit.analyzers\test\xunit.analyzers.tests\xunit.analyzers.tests.csproj]
C:\Program Files\dotnet\sdk\3.1.100-preview2-014569\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(255,5): error MSB4018:    at Microsoft.NET.Build.Tasks.GenerateRuntimeConfigurationFiles.AddFrameworks(RuntimeOptions runtimeOptions, ProjectContext projectContext) [C:\dev\github\xunit.analyzers\test\xunit.analyzers.tests\xunit.analyzers.tests.csproj]
C:\Program Files\dotnet\sdk\3.1.100-preview2-014569\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(255,5): error MSB4018:    at Microsoft.NET.Build.Tasks.GenerateRuntimeConfigurationFiles.WriteRuntimeConfig(ProjectContext projectContext) [C:\dev\github\xunit.analyzers\test\xunit.analyzers.tests\xunit.analyzers.tests.csproj]
C:\Program Files\dotnet\sdk\3.1.100-preview2-014569\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(255,5): error MSB4018:    at Microsoft.NET.Build.Tasks.GenerateRuntimeConfigurationFiles.ExecuteCore() [C:\dev\github\xunit.analyzers\test\xunit.analyzers.tests\xunit.analyzers.tests.csproj]
C:\Program Files\dotnet\sdk\3.1.100-preview2-014569\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(255,5): error MSB4018:    at Microsoft.NET.Build.Tasks.TaskBase.Execute() [C:\dev\github\xunit.analyzers\test\xunit.analyzers.tests\xunit.analyzers.tests.csproj]
C:\Program Files\dotnet\sdk\3.1.100-preview2-014569\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(255,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [C:\dev\github\xunit.analyzers\test\xunit.analyzers.tests\xunit.analyzers.tests.csproj]
C:\Program Files\dotnet\sdk\3.1.100-preview2-014569\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(255,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [C:\dev\github\xunit.analyzers\test\xunit.analyzers.tests\xunit.analyzers.tests.csproj]
```

**Comment by nguerrera** [comment]
@livarcocc @vitek-karas @dsplaisted

**Comment by vitek-karas** [comment]
#3859 is the proposed fix.

That said the project should never set `GenerateRuntimeConfigFiles` to true for anything but .NET Core TFMs. Generating `.runtimeconfig.json` for .NET Framework TFM doesn't make sense and the content of the file is basically useless.

This: https://github.com/xunit/xunit.analyzers/blob/74f6d366bd5a33f7ed966fd6f1fb7703452dac0d/test/Directory.build.props#L11 should be changed to something like:
``` XML
    <GenerateRuntimeConfigurationFiles Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'">true</GenerateRuntimeConfigurationFiles>
```

With that change the project builds just fine even on 3.1.1xxx.

**Comment by nguerrera** [comment]
That would need to be moved to Directory.Build.targets as TargetFrameworkIdentifier isn't set when props are evaluated

**Comment by vitek-karas** [comment]
#3859 got merged to 3.1.1xxx - should be fixed then.
